File: modality-network-datastore/tests/models/round.rs
#[cfg(test)]
mod tests {
    use anyhow::{Result};
    use modality_network_datastore::NetworkDatastore;
    use modality_network_datastore::models::round::Round;
    use modality_network_datastore::models::round::prelude::*;
    
    #[tokio::test]
    async fn test_round() -> Result<()> {
        let datastore = NetworkDatastore::create_in_memory()?;
        
        let round = Round::create_from_json(serde_json::json!({"round": 1}))?;
        round.save(&datastore).await?;
        
        let round = Round::create_from_json(serde_json::json!({"round": 2}))?;
        round.save(&datastore).await?;
        
        let round = Round::create_from_json(serde_json::json!({"round": 3}))?;
        round.save(&datastore).await?;
        
        let max_round = Round::find_max_id(&datastore).await?;
        assert_eq!(max_round, Some(3));

        Ok(())
    }

    #[test]
    fn test_add_remove_scribe() -> Result<()> {
        let mut round = Round::create_from_json(serde_json::json!({"round": 1}))?;
        
        round.add_scribe("peer1".to_string());
        assert_eq!(round.scribes, vec!["peer1"]);

        round.add_scribe("peer2".to_string());
        assert_eq!(round.scribes, vec!["peer1", "peer2"]);

        round.remove_scribe("peer1");
        assert_eq!(round.scribes, vec!["peer2"]);

        Ok(())
    }
}

File: js/packages/network-datastore/src/data/Round.js
import Model from "./Model.js";
export default class Round extends Model {
  static id_path = "/consensus/round/${round}";
  static fields = ["round", "scribes"];
  static field_defaults = {
    scribes: [],
  };

  static findMaxId({ datastore }) {
    return datastore.findMaxIntKey(`/consensus/round`);
  }

  addScribe(scribe_peer_id) {
    this.scribes.push(scribe_peer_id);
  }

  removeScribe(scribe_peer_id) {
    this.scribes = this.scribes.filter((s) => s !== scribe_peer_id);
  }
}
